presto-main/src/test/java/com/facebook/presto/sql/query/QueryAssertions.java

```diff
@@ -34,21 +34,27 @@
         implements Closeable
 {
     private final QueryRunner runner;
+    private Session session;
 
     public QueryAssertions()
     {
-        Session defaultSession = testSessionBuilder()
+        session = testSessionBuilder()
                 .setCatalog("local")
                 .setSchema("default")
                 .build();
 
-        runner = new LocalQueryRunner(defaultSession);
+        runner = new LocalQueryRunner(session);
+    }
+
+    public void setSession(Session session)
```

Comment:
Why not to pass session to constructor? Notice that your approach is vulnerable, as one may set session in test and that would make the other test to fail. Also it is not thread safe, tests cannot be run in parallel.

Reply:
Another left-over from an experiment I was working on where I needed to do it this way. I'll fix it.